### PR TITLE
LF: enforce non-empty maintainer in contracts key.

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This fixes a bug in the Speedy interpreter.  With this change, the
interpreter crashes if it attempts to create a contract containing a 
contract key that has an empty set of maintainers.

This is a breaking change approved by @bame-da. 

CHANGELOG_BEGIN
- [LF] (Bug fix) enforce non-empty maintainers in contract key during
  contract creation.

CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/EmptyContractKeyMaintainers.daml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import qualified DA.Next.Set as Set
+import DA.Next.Set (Set)
+
+template NoMaintainer
+  with
+    sig: Party
+  where
+    signatory sig
+    key sig : Party
+    maintainer (Set.empty : Set Party)
```
Comment (hurryabit) [inline compiler/damlc/tests/daml-test-files/EmptyContractKeyMaintainers.daml:13]: We can use lists of parties here as well. That would make it simpler.
```suggestion
    maintainer []
```

File: daml-lf/tests/BasicTests.daml [hunk 2]
```diff
@@ -589,3 +592,11 @@ template ComputeContractKeyWhenExecutingCreate with
...
         let _ignore = create ComputeContractKeyAfterEnsureClause with owner
         pure ()
+
+template NoMaintainer
+  with
+    sig: Party
+  where
+    signatory sig
+    key sig : Party
+    maintainer (Set.empty : Set Party)
```
Comment (hurryabit) [inline daml-lf/tests/BasicTests.daml:602]: Same as above.
```suggestion
    maintainer []
```

**Review by hurryabit — APPROVED** [review]
Provided that @bame-da is fine with this breaking change, which he seems to be, I'm fine too.

Thanks a lot for fixing this!

File: compiler/damlc/tests/daml-test-files/EmptyContractKeyMaintainers.daml (hunk 1 above)
Comment (remyhaemmerle-da) [reply compiler/damlc/tests/daml-test-files/EmptyContractKeyMaintainers.daml]: done

File: daml-lf/tests/BasicTests.daml (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/tests/BasicTests.daml]: changed.

**Pull request merged**